src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/AttackSystem.java

```diff
@@ -76,4 +85,39 @@ public void onActivate(ActivateEvent event, EntityRef entity) {
             }
         }
     }
+
+    @Override
+    public void update(float delta) {
```

Comment:
This is a fairly intensive check to run every single frame. Can we listen to the `InventorySlotChangedEvent` instead and then check if the new item is a flag?
Then if it is we handle the team cases